akka-docs/rst/java/code/docs/stream/javadsl/cookbook/RecipeGlobalRateLimit.java

```diff
@@ -121,11 +119,12 @@ private Limiter(int maxAvailableTokens, FiniteDuration tokenRefreshPeriod,
 
     private void releaseWaiting() {
       final List<ActorRef> toBeReleased = new ArrayList<>(permitTokens);
-      for (int i = 0; i < permitTokens && i < waitQueue.size(); i++) {
-        toBeReleased.add(waitQueue.remove(i));
+      for (Iterator<ActorRef> it = waitQueue.iterator(); permitTokens > 0 && it.hasNext();) {
+          toBeReleased.add(it.next());
+          it.remove();
+          permitTokens --;
       }
 
-      permitTokens -= toBeReleased.size();
```

Comment:
I meant "for loop", not only this line. By the way how can i select line range to comment?